utilities/ImpactUtilities.groovy

```diff
@@ -867,4 +878,48 @@ def sortFileList(list) {
  */
 def isMappedAsZUnitConfigFile(String file) {
 	return (dependencyScannerUtils.getScanner(file).getClass() == com.ibm.dbb.dependency.ZUnitConfigScanner)
+}
+
+/*
+ *  addLinkDependencies -
+ *  method to identify all statically called sub module programs when the main program changes
+ *
+ *  @return list of statically called sub modules
+ *
+ */
+def addLinkDependencies(buildSet, buildLinkSet) {
+    String OutputsCollectionName = props.applicationOutputsCollectionName
+    MetadataStore metadataStore = MetadataStoreFactory.getMetadataStore()
+
+    buildSet.each { file ->
+        def logicalFile = buildUtils.relativizePath(file)
+        def logicalFiles = metadataStore.getCollection(props.applicationOutputsCollectionName).getLogicalFile(logicalFile)
+
+        if (logicalFiles) {
+            // Check if any logical files are found
+            // List all link dependencies for every main program that changes from the output collection. 
+            // This will return only the program name and not the absolute path
+            logicalFiles.each { logicalFileRecord ->
+                def dependencies = logicalFileRecord.getLogicalDependencies()
+
+                dependencies.each { logicalDep ->
+                    def linkDepName = logicalDep.getLname()
+                    def linkDepLogicalFile = metadataStore.getCollection(props.applicationCollectionName).getLogicalFiles(linkDepName)
+
+                    // Get the logical path for all the link dependencies returned
+                    linkDepLogicalFile.each { filePath ->
+                        // Link Dependency Files to be added
+                        def linkDepFile = filePath.getFile()
+
+                        if (linkDepFile != logicalFile) {
+                            buildLinkSet.add(linkDepFile)
```

Comment:
We should only add files, that have a ScriptMapping ( so it is a proper sub-module)
```
if (ScriptMappings.getScriptName(linkDepFile)) ... 
```

Reply:
check included